packages/@aws-cdk/aws-ivs/lib/playback-key-pair.ts

```diff
@@ -0,0 +1,72 @@
+import * as core from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnPlaybackKeyPair } from './ivs.generated';
+
+/**
+ * Represents an IVS Playback Key Pair
+ */
+export interface IPlaybackKeyPair extends core.IResource {
+  /**
+   * Key-pair ARN. For example: arn:aws:ivs:us-west-2:693991300569:playback-key/f99cde61-c2b0-4df3-8941-ca7d38acca1a
+   *
+   * @attribute
+   */
+  readonly playbackKeyPairArn: string;
+}
+
+/**
+ * Reference to a new or existing IVS Playback Key Pair
+ */
+abstract class PlaybackKeyPairBase extends core.Resource implements IPlaybackKeyPair {
+  // these stay abstract at this level
+  public abstract readonly playbackKeyPairArn: string;
+}
+
+/**
+ * Properties for creating a new Playback Key Pair
+ */
+export interface PlaybackKeyPairProps {
+  /**
+   * The public portion of a customer-generated key pair.
+   */
+  readonly publicKeyMaterial: string;
+
+  /**
+   * An arbitrary string (a nickname) assigned to a playback key pair that helps the customer identify that resource.
+   * The value does not need to be unique.
+   *
+   * @default Empty String
+   */
+  readonly name?: string;
+}
+
+/**
+  A new IVS Playback Key Pair
+*/
+export class PlaybackKeyPair extends PlaybackKeyPairBase {
+  public readonly playbackKeyPairArn: string;
+
+  /**
+  * Key-pair identifier. For example: 98:0d:1a:a0:19:96:1e:ea:0a:0a:2c:9a:42:19:2b:e7
+  *
+  * @attribute
+  */
+  public readonly fingerprint: string;
```

Comment:
Similar to the `name` props, the attributes are prefixed with the type name; I'm actually really surprised you're not getting a linter exception here. Should be `playbackKeyPairFingerprint`.

Reply:
I thought the lint rule was specifically for resources with a physicalName. None of these resources require a name so I thought that meant that they weren't resources with a physicalName. I can change all these if it makes sense, it's just not really that obvious that I needed to do it.